File: character-counter/queue.rb
class Queue
    attr_accessor :items

    def initialize
        @items = []
    end

    def enqueue(element)
        items.push(element)
    end


    def dequeue()
        items.shift
    end

    def front
        items[0]
    end

    def empty?
        return items.length == 0
    end

    def show
        str = ""
        for item in items do
            str += item + " "
        end
        return str
    end
end

q = Queue.new
q.enqueue("First In")
q.enqueue("Second In")
q.enqueue("Third In")
puts q.show

puts q.dequeue()
puts q.show()

File: character-counter/queue.js
class Queue{
    constructor(){
      this.items = []
    }
    enqueue(item){
      this.items.push(item)
    }
    
    front(){
      return this.items[0]
    }
    
    dequeue(){
      return this.items.shift()
    }
    
    isEmpty(){
      return this.items.length == 0
    }
    
    show(){
      var str = "";
      for (var i = 0; i < this.items.length; i++)
          str += this.items[i] + " ";
      return str;
    }
  }
  
  let q = new Queue()
  q.enqueue("First")
  q.enqueue("Second")
  q.enqueue("Third")
  
  console.log(q.dequeue())
  console.log( q.show())
  console.log(q.front())